sim/core/buffs.go

```diff
@@ -55,9 +55,11 @@ func applyBuffEffects(agent Agent, raidBuffs proto.RaidBuffs, partyBuffs proto.P
 		character.AddStat(stats.SpellCrit, 5*CritRatingPerCritChance)
 	}
 	if raidBuffs.MoonkinAura == proto.TristateEffect_TristateEffectImproved || raidBuffs.SwiftRetribution {
-		character.PseudoStats.CastSpeedMultiplier *= 1.03
-		character.PseudoStats.MeleeSpeedMultiplier *= 1.03
-		character.PseudoStats.RangedSpeedMultiplier *= 1.03
+		hastePercentBonus := 3.0
+		character.AddStats(stats.Stats{
+			stats.SpellHaste: hastePercentBonus*HasteRatingPerHastePercent,
+			stats.MeleeHaste: hastePercentBonus*HasteRatingPerHastePercent,
+		})
 	}
```

Comment:
This breaks DKs ghoul pet scaling as it scales with its owners haste rating. Having the buff give haste rating means the pet will first get the the bonus from the buff on the owner then get it again when the buff is applied to it (permanent pets receive raid buffs)

Why did you need to make this change?

Why do you need this change?

Reply:
I needed to change it because it now fits the buff description and now you can actually correctly see it in the UI.
Sadly, there's no way around that since "CastSpeedMultiplier" is different from Haste.

Reply:
Don't change internals like this just to satisfy something in the UI; this is going to change actual DPS values, as the haste stat and speed multipliers do not work the same way.

We can chat about adding something to make this visible in UI but it should be separate.

Reply:
> I needed to change it because it now fits the buff description

It does not fit the buff description:
![image](https://user-images.githubusercontent.com/5707996/180654618-2c26761d-444a-417c-a79d-f0109b05a917.png)

Its previous implementation is the correct one imo

Reply:
So none of these buffs change haste? My bad if that's the case, I will revert these changes. I thought the spell tooltip was correct.

Reply:
Oh I see. They both change haste, but if it's through rating, it becomes inheritable?